composer.json

```diff
@@ -34,7 +34,8 @@
         "amphp/socket": "^1",
         "amphp/sync": "^1.3",
         "league/uri": "^6",
-        "psr/http-message": "^1"
+        "psr/http-message": "^1",
+        "laminas/laminas-diactoros": "^2.3"
```

Comment:
Please move this back to require-dev, it should remain an optional feature.

Reply:
Now that's impossible because adapter uses Request object from Laminas library. I've tried to find League implementation but they use Laminas in their examples; so Laminas is not optional anymore.

Reply:
We can avoid requiring concrete PSR-7 implementation, though, if we accept RequestInterface in corresponding method (and then call with* methods). But the problem is body - it's mutable in PSR-7.

Reply:
None of the added classes are mandatory to use the http-client. If one doesn't use any of them (I don't), one shouldn't be force to install a dead-code dependency. Moving it back to require-dev fixes the issue.

About diactoros, could it be possible to be compatible with nyholm/psr7? That's the lib we should recommend IMHO (I don't know about the policy of the amp project here, I'm just talking for Symfony-related apps.)

Reply:
Wait, I'll try one idea...

Reply:
I've got rid of requiring Laminas with injecting PSR-7 `RequestFactoryInterface`. Maybe it would be more convenient to inject it in constructor - but `from*` methods will never use it, so in this case we'll have to split adapter into independent from* and to* parts.

Reply:
On the other hand, most use cases will require using either  "fromPsrRequest/toPsrResponse" or "toPsrRequest/fromPsrResponse" pair of methods. One of them will require `RequestFactoryInterface`, so I'll better inject it in constructor.